### PR TITLE
GEOMESA-2513 Catch errors in deserialization of corrupt data

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/serialization/TwkbSerialization.scala [hunk 1]
```diff
@@ -48,7 +51,7 @@ trait TwkbSerialization[T <: NumericWriter, V <: NumericReader]
         val coord = geometry.getCoordinate
         // check for dimensions - use NaN != NaN to verify z coordinate
```
Comment (jnh5y) [inline geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/serialization/TwkbSerialization.scala]: Update comment?

**Comment by jnh5y** [comment]
@elahrvivaz have you checked to see if the try-catchs are slowing things done?

**Comment by elahrvivaz** [comment]
trys that don't throw should be essentially free: https://stackoverflow.com/questions/16451777/is-it-expensive-to-use-try-catch-blocks-even-if-an-exception-is-never-thrown

File: geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/serialization/TwkbSerialization.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/serialization/TwkbSerialization.scala]: oops, sorry didn't see this. I think it's still somewhat valid as that's what `Double.isNaN` does anyway

**Pull request merged**